src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/output/PricesSpreadsheetGenerator.kt

```diff
@@ -1,25 +1,33 @@
 package uk.gov.justice.digital.hmpps.pecs.jpc.output
 
+import org.apache.poi.ss.usermodel.Workbook
 import org.apache.poi.xssf.usermodel.XSSFWorkbook
+import org.slf4j.LoggerFactory
 import org.springframework.beans.factory.annotation.Autowired
 import org.springframework.beans.factory.annotation.Qualifier
 import org.springframework.stereotype.Component
-import uk.gov.justice.digital.hmpps.pecs.jpc.calculator.MovePrice
+import uk.gov.justice.digital.hmpps.pecs.jpc.calculator.PriceCalculator
 import uk.gov.justice.digital.hmpps.pecs.jpc.pricing.Supplier
+import uk.gov.justice.digital.hmpps.pecs.jpc.reporting.MoveFiltererParams
 import java.io.File
 import java.io.FileOutputStream
 import java.time.Clock
 import java.time.LocalDate
 
 @Component
 class PricesSpreadsheetGenerator(@Autowired @Qualifier(value = "spreadsheet-template") private val template: File,
-                                 @Autowired private val clock: Clock) {
+                                          @Autowired private val clock: Clock) {
 
-    fun generateSpreadsheet(from: LocalDate, to: LocalDate, supplier: Supplier, prices: Sequence<MovePrice>): File {
-        val dateRun = LocalDate.now(clock)
+    private val logger = LoggerFactory.getLogger(javaClass)
+
+    internal fun generate(from: LocalDate, to: LocalDate, supplier: Supplier, calculator: PriceCalculator): File {
```

Comment:
from, to, supplier look like the same as in MoveFiltererParams, so we should probably use this instead. Likewise, we probably don't need a dedicated PriceSheet.DateRange, that could also be within the MoveFiltererParams (which we could rename to make more generic). We could use a type of ClosedRange<LocalDate> to represent the date range.

Reply:
Yeah agreed.  The date range this is a bit of a hangover and doesn't add much.

Reply:
Have added a ClosedRangeLocalDate see what you think.  Not doe a change to the MoveFiltererParams.  Perhaps leave/do separately as may get a clash.